pybamm/experiment/experiment.py

```diff
@@ -129,6 +99,27 @@ def __init__(
         self.termination_string = termination
         self.termination = self.read_termination(termination)
 
+    @staticmethod
+    def process_steps(unprocessed_steps, period, temp):
+        processed_steps = {}
+        for step in unprocessed_steps:
+            if repr(step) in processed_steps:
+                continue
+            elif isinstance(step, str):
+                processed_step = pybamm.step.string(step)
+            elif isinstance(step, pybamm.step._Step):
+                processed_step = step
+            else:
+                raise TypeError("Operating conditions must be a Step object or string.")
+
+            if processed_step.period is None:
+                processed_step.period = period
+            if processed_step.temperature is None:
+                processed_step.temperature = temp
+
+            processed_steps[repr(step)] = processed_step
+        return processed_steps
```

Comment:
@rtimms I extracted this method so it was easily testable. I added a test for it.